### DISCUSSION TITLE
FOUC - tailwind + remix

### DISCUSSION THREAD

**Discussion by dan-cooke** [remix-run/remix#4747] [Q&A] [answered]
### What version of Remix are you using?

1.8.0

### Steps to Reproduce

I filed this over at tailwindcss, too - here is the full issue https://github.com/tailwindlabs/tailwindcss/issues/9990

**What version of Tailwind CSS are you using?**
3.2.4

**What build tool (or framework if it abstracts the build tool) are you using?**
```
 "dependencies": {
    "@remix-run/node": "^1.8.0",
    "@remix-run/react": "^1.8.0",
    "@remix-run/serve": "^1.8.0",
    "isbot": "^3.6.5",
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "devDependencies": {
    "@remix-run/dev": "^1.8.0",
    "@remix-run/eslint-config": "^1.8.0",
    "@types/react": "^18.0.25",
    "@types/react-dom": "^18.0.8",
    "autoprefixer": "^10.4.13",
    "concurrently": "^7.6.0",
    "eslint": "^8.27.0",
    "postcss": "^8.4.19",
    "tailwindcss": "^3.2.4",
    "typescript": "^4.8.4"
  },
```

**What version of Node.js are you using?**
16.13.0

**What browser are you using?**
Chrome

**What operating system are you using?**

macOS

**Reproduction URL**

https://github.com/dan-cooke/remix-tailwind-bug

**Describe your issue**

Simply follow the [remix setup example here](https://tailwindcss.com/docs/guides/remix) .

And running `npm run dev`

You get a FOUC.

Here is a screenshot of the server render.
![image](https://user-images.githubusercontent.com/22816887/205326587-2d810863-b830-46ef-9d77-9e0aef1037b8.png)

HEre is the browser render, 1 frame after the hydration
![image](https://user-images.githubusercontent.com/22816887/205326682-3d636985-053c-4677-95da-0907d20fd99a.png)

### Hydration errors
This obviously is causing hydration errors as the server output is much different
```
Uncaught Error: Hydration failed because the initial UI does not match what was rendered on the server.
```

### Expected Behavior

<>

### Actual Behavior

/

**Comment by machour** [comment, marked as answer]
There is no rendering tailwind on the server.
Tailwind generates your CSS in a different process, and Remix just renders the `<link>` tag.
You probably have an extension messing up hydration (see https://remix.run/docs/en/v1/pages/gotchas#browser-extensions-injecting-code)

Converting this to a Q&A discussion.

**Reply by dan-cooke** [reply]
I see. So how exactly are you supposed to prevent a FOUC with Remix + Tailwind?

PS: you are right about browser extension causing hydration errors, i was not on my work chrome profile
